nextflow.config

```diff
@@ -74,7 +74,15 @@ profiles {
       	params.workdir = "/beegfs/rna-hta/$USER/work"
   	    params.permanentCacheDir = "/beegfs/rna-hta/nextflow-rnaseq-autodownload/"
         conda.cacheDir = "/beegfs/rna-hta/$USER/nextflow-conda-cache"
+        process {
+            withLabel: smallTask { executor = 'local' }
+            withLabel: hisat2 { cpus = 48 ; memory = { 70.GB * task.attempt } ; maxRetries = 3 ; errorStrategy = { task.exitStatus in 134..140 ? 'retry' : 'terminate' } ; clusterOptions = '--partition=s_standard,s_fat,b_standard,b_fat' }
+            withLabel: fastp { cpus = 12 ; memory = '30 GB' ; clusterOptions = '--partition=s_standard,s_fat,b_standard,b_fat' }
+            withLabel: subread { cpus = 12 ; memory = '30 GB' ; clusterOptions = '--partition=s_standard,s_fat,b_standard,b_fat' }
+            withLabel: sortmerna { cpus = 12 ; memory = '30 GB' ; clusterOptions = '--partition=s_standard,s_fat,b_standard,b_fat' }
+        }
+
         includeConfig 'configs/slurm.config'
```

Comment:
ah okay... correct me but I thought we have these definitions in a separate  `*.config` file? or is this only added for the ara profile specifically here? 

And: do you know how the 'override' works? So let's say `configs/slurm.config` defines `cpus` for `hisat2` - which will be used then? do configurations in `nextflow.config` always count over definitions in other config files?

Reply:
right now, we just set `cpus` in local.config - and this stuff in the ara, but not slurm, profile. 

maybe this is also necessary in the `slurm.config`, so that the slurm profile works also as stand-alone profile?
 
---

the [documentations](https://www.nextflow.io/docs/latest/config.html#config-profiles) warns: 
> When using the profiles feature in your config file do NOT set attributes in the same scope both inside and outside a profiles context

I'm not sure if I do this here, because `slurm.config` has also a process scope

and I have no idea, how the profiles are merges. [Here](https://www.nextflow.io/docs/latest/config.html#config-profiles) is something documented, but I'm not sure what happens when two 'withLabel' are set

Reply:
I think the slurm profile should be also able to run stand-alone.

Yes, I am also not sure how the merging works exactly. But maybe lets have a look here:
https://github.com/replikation/What_the_Phage

Here I thought together with Christian about a distinction of "environment" and "engine":

1) Choose the environment: `local`, `lsf` or `ebi`
2) Choose the engine: `docker` or `singularit`y

examples:
* `-profile local,docker`
* `-profile local,singularity`
* `-profile lsf,docker`

so in our case we would like to have:

* `-profile local,conda`
* `-profile slurm,conda`
* `-profile ara,conda`

I think this is really clean then? Maybe you can have a look how Christian implemented this in WtP and adapt this idea?

Reply:
right now we kind of have also this distinction:
- `local` and `slurm` as executer
- `conda` as engine

Our standard/default setting is `local` + `conda` and with `-profile ara` we have some ARA specific paths and setting and `slurm` + `conda`

I'd just add `cpus` and `memory` settings in `slurm`. (And test if we can overwrite/change these settings in the ARA profile)

Reply:
true, I think thats fine because we anyway don't want to run something like `-profile slurm,docker` at the moment.

Reply:
we have now `cpus, `memory` (and error handling for HISAT2) defined in `slurm`. `ara` adds the local executor to the `smallTask` processes.

the used process directives in `slurm` and `ara` are non-overlapping, so  there is no overwriting

Reply:
Ok great, looks good, I think we are ready to merge then